lessons/modelevaluation.ipynb

```diff
@@ -0,0 +1,786 @@
+   },
+   "source": [
+    "**Posterior predictive model checking** - logic:\n",
+    "* If our model is the true one, then *replica* data generated by it should \"look like\" the one dataset we have. \n",
+    "\n",
+    "* This means that any *summary* $T$ of both the real dataset, $T(d)$, and the replica datasets, $T(d^{\\rm rep})$, should follow the same distribution over noise realizations _and_ model parameters.\n",
+    "\n",
+    "* If the real dataset was not generated with our model, then its summary may be an _outlier_ from the distribution of summaries of replica datasets."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Visual comparison of _models_ drawn from the posterior with the data. (NB: there is no replica data here...)\n",
+    "<table>\n",
+    "    <tr>\n",
+    "        <td><img src=\"graphics/modelcheck-linear-posterior.png\" width=90%></td>\n",
+    "        <td></td>\n",
+    "        <td><img src=\"graphics/modelcheck-linear.png\" width=90%></td>\n",
+    "    </tr>\n",
+    "</table>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Example test statistic: Pearson Correlation $r_{12}$\n",
+    "\n",
+    "* Focuses on the tightness of linear correlation between $x$ and $y$\n",
+    "\n",
+    "* $T(d) = r_{12} = \\frac{\\sum_i (x_i - \\bar{x})(y_i - \\bar{y})}{\\left[ \\sum_i (x_i - \\bar{x})^2 \\sum_i (y_i - \\bar{y})^2 \\right]^{1/2}}$\n",
+    "\n",
+    "For each one of many posterior samples, we draw a _replica dataset_ from the sampling distribution given the sample parameter vector, and compute $T(d^{\\rm rep})$, building up a histogram of $T$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "${\\rm P}[T(d^{\\rm rep})>T(d)\\,|\\,d] = 99.43\\%$ - our dataset $d$ is clearly an outlier.\n",
+    "\n",
+    "<img src=\"graphics/modelcheck-linear-TS.png\" width=50%>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "* The posterior predictive probability distribution for the test statistic $T(d)$ generated by sampling in this way is marginalized over both parameters and (replica) datasets \n",
+    "\n",
+    "* It takes into account both the uncertainty in the data (captured by the sampling distribution) _and_ the uncertainty in the parameters (propagated from our one dataset and our prior knowledge during posterior sampling)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "* Posterior predictive model checking can be seen as the Bayesian extension of classical hypothesis testing, and is a useful test of _model adequacy_ \n",
```

Comment:
Might be good to explicitly note the similarity in logic to the chisquare test earlier

Reply:
✅